docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -78,15 +78,248 @@ If you need to have external labels in your query results, this is currently not
-> Please refer to [`thanos tools bucket rewrite` documentation](https://thanos.io/tip/components/tools.md/#bucket-rewrite) for more details.
-
 **Grafana Mimir does not support Thanos’ _downsampling_ feature.**
 To guarantee query results correctness please only upload original (raw) Thanos blocks into Mimir's storage.
 If you also upload blocks with downsampled data (ie. blocks with non-zero `Resolution` field in `meta.json` file), Grafana Mimir will merge raw samples and downsampled samples together at the query time.
 This may cause that incorrect results are returned for the query.
 
-> **Note**: It is possible to run compaction and deduplicate blocks by Thanos first by using `thanos compact` command
-> with `--compact.enable-vertical-compaction --deduplication.func=penalty --deduplication.replica-label=<LABEL>` flags.
-> Please refer to [Vertical Compaction Use Cases](https://thanos.io/tip/components/compact.md/#vertical-compaction-use-cases) for more details on offline deduplication.
+## Migrate historic TSDB blocks from Thanos to Grafana Mimir
+
+1) Copy the blocks from Thanos's bucket to an Intermediate bucket
+
+   Create an intermediate object storage bucket(S3,GCS,etc.) in your cloud provider where you can copy the historical blocks and work on them before uploading to the Mimir bucket.
+   
+   > **Note**: It is recommeneded to run all of the following steps in a **screen** or **tmux** command to avoid any interruptions as these may take time depending on the amount of data to be processed.
+
+    For AWS S3, use the aws tool in the following command:
+
+    ```bash
+    aws s3 cp -r s3://<THANOS-BUCKET> s3://<TENANT>/<DIRECTORY>
+    ```
+
+    For Google Cloud Storage (GCS), use the gsutil tool in the following command:
+    
+    ```bash
+    gsutil -m cp -r gs://<THANOS-BUCKET>/* gs://<INTERMEDIATE-MIMIR-BUCKET>/
+    ```
+    
+    Once the copy process is done, inspect the blocks in the bucket to ensure it's valid from Thanos perspective.
+    
+    ```bash
+    thanos tools bucket inspect \
+        --objstore.config-file bucket.yaml
+    ```
+
+2) Remove the downsampled blocks
+
+   Mimir doesn't understand the downsampled blocks from Thanos (ie. blocks with non-zero `Resolution` field in meta.json file). Therefore we need to remove the 5m and 1h downsampled blocks from this bucket.
+   
+   Mark the downsampled blocks for deletion.
+   
+   ```bash
+   thanos tools bucket retention \
+       --objstore.config-file bucket.yaml \
+       --retention.resolution-1h=1s \
+       --retention.resolution-5m=1s \
+       --retention.resolution-raw=0s
+   ```
+   Cleanup the blocks marked for deletion.
+   
+   ```bash
+   thanos tools bucket cleanup \
+       --objstore.config-file bucket.yaml \
+       --delete-delay=0
+   ```
+  
+3) Remove the duplicated blocks
+
+   If two replicas of Prometheus instances are deployed for HA, then only upload the blocks from one of the replicas and drop the other.
```

Comment:
```suggestion
   If two replicas of Prometheus instances are deployed for high-availability, then only upload the blocks from one of the replicas and drop the other.
```

Reply:
@pstibrany, the other block, right? Please vet for technical accuracy.

Reply:
It's other replica.